File: OfflineFetcher/app/src/main/java/de/jbamberger/offlinefetcher/ui/reddit/RedditSubcomponent.java
package de.jbamberger.offlinefetcher.ui.reddit;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@RedditScope
@Subcomponent(modules = {RedditModule.class})
public interface RedditSubcomponent extends AndroidInjector<RedditActivity> {
    @Subcomponent.Builder
    public abstract class Builder extends AndroidInjector.Builder<RedditActivity> {}
}

File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/ui/reddit/RedditSubComponent.kt
package de.jbamberger.offlineclient.ui.reddit

import dagger.Subcomponent
import dagger.android.AndroidInjector

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@RedditScope
@Subcomponent(modules = [RedditModule::class])
interface RedditSubComponent : AndroidInjector<RedditActivity> {

    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<RedditActivity>()
}
